### PR TITLE
En_Zot

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Zot/z_en_zot.c [hunk 1]
```diff
@@ -22,123 +57,1366 @@ const ActorInit En_Zot_InitVars = {
...
+    return false;
+}
+
+void func_80B97100(EnZot* this, GlobalContext* globalCtx) {
+}
+
+void func_80B97110(EnZot* this, GlobalContext* globalCtx) {
+    u16 textId;
+
+    if (gSaveContext.playerForm == 2) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Zot/z_en_zot.c:375]:
```suggestion
    if (gSaveContext.playerForm == PLAYER_FORM_ZORA) {
```

File: src/overlays/actors/ovl_En_Zot/z_en_zot.c [hunk 2]
```diff
@@ -22,123 +57,1366 @@ const ActorInit En_Zot_InitVars = {
...
+               Actor_IsActorFacingLink(&this->actor, 0x3000)) {
+        func_800B8614(&this->actor, globalCtx, 120.0f);
+    }
+}
+
+void func_80B972E8(EnZot* this, GlobalContext* globalCtx) {
+    u16 textId;
+
+    if (gSaveContext.weekEventReg[29] & 0x10) {
+        if (gSaveContext.playerForm == 2) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Zot/z_en_zot.c:426]:
```suggestion
        if (gSaveContext.playerForm == PLAYER_FORM_ZORA) {
```

File: src/overlays/actors/ovl_En_Zot/z_en_zot.c [hunk 3]
```diff
@@ -22,123 +57,1366 @@ const ActorInit En_Zot_InitVars = {
...
+            }
+        } else {
+            textId = 0x1267;
+            if (gSaveContext.weekEventReg[29] & 2) {
+                textId = 0x1269;
+            } else {
+                gSaveContext.weekEventReg[29] |= 2;
+            }
+        }
+    } else if (gSaveContext.playerForm == 2) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Zot/z_en_zot.c:441]:
```suggestion
    } else if (gSaveContext.playerForm == PLAYER_FORM_ZORA) {
```

File: src/overlays/actors/ovl_En_Zot/z_en_zot.c [hunk 4]
```diff
@@ -22,123 +57,1366 @@ const ActorInit En_Zot_InitVars = {
...
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->actionFunc = func_80B973BC;
+    } else {
+        func_800B8500(&this->actor, globalCtx, 10000.0f, 1000.0f, 0);
+    }
+}
+
+void func_80B9765C(EnZot* this, GlobalContext* globalCtx) {
+    if (1) {
+        do { } while (0); }
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Zot/z_en_zot.c:537]: Did you forgot to run the formatter? Or the formatter thought this was a good idea?

File: src/overlays/actors/ovl_En_Zot/z_en_zot.c [hunk 5]
```diff
@@ -22,123 +57,1366 @@ const ActorInit En_Zot_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zot/func_80B98BF4.s")
+    gfx -= 2;
+    gfxCtx->polyOpa.d = gfx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zot/func_80B98CA8.s")
+    gSPEndDisplayList(gfx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zot/func_80B98E10.s")
+    return gfx;
+}
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Zot/z_en_zot.c:1367]: This is the `GRAPH_ALLOC` macro. You can use it like this:

```c
Gfx* func_80B99580(GraphicsContext* gfxCtx) {
    Gfx* dList;

    dList = GRAPH_ALLOC(gfxCtx, sizeof(Gfx) * 2);
    gSPEndDisplayList(dList);
    return dList;
}
```

File: src/overlays/actors/ovl_En_Zot/z_en_zot.c [hunk 6]
```diff
@@ -22,123 +57,1366 @@ const ActorInit En_Zot_InitVars = {
...
+    if (path == NULL) {
+        return true;
+    }
+
+    points = &((Vec3s*)Lib_SegmentedToVirtual(path->points))[this->unk_2D4];
+    temp_f12 = points->x - this->actor.world.pos.x;
+    temp_f14 = points->z - this->actor.world.pos.z;
+    this->actor.world.rot.y = Math_Atan2S(temp_f12, temp_f14);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, 2000, 200);
+    phi_f2 = SQ(this->actor.speedXZ) * 9.0f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Zot/z_en_zot.c:314]:
```suggestion
    phi_f2 = SQ(this->actor.speedXZ) * SQ(3.0f);
```

File: src/overlays/actors/ovl_En_Zot/z_en_zot.c [hunk 7]
```diff
@@ -22,123 +57,1366 @@ const ActorInit En_Zot_InitVars = {
...
+
+    if (this->unk_2D4 == 0) {
+        phi_f2 = 400.0f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Zot/z_en_zot.c:317]:
```suggestion
        phi_f2 = SQ(20.0f);
```

File: src/overlays/actors/ovl_En_Zot/z_en_zot.c [hunk 8]
```diff
@@ -22,123 +57,1366 @@ const ActorInit En_Zot_InitVars = {
...
+        return true;
+    }
+
+    points = &((Vec3s*)Lib_SegmentedToVirtual(path->points))[this->unk_2D4];
+    temp_f12 = points->x - this->actor.world.pos.x;
+    temp_f14 = points->z - this->actor.world.pos.z;
+    this->actor.world.rot.y = Math_Atan2S(temp_f12, temp_f14);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, 2000, 200);
+
+    if ((SQ(temp_f12) + SQ(temp_f14)) < 100.0f) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Zot/z_en_zot.c:348]:
```suggestion
    if ((SQ(temp_f12) + SQ(temp_f14)) < SQ(10.0f)) {
```

**Review by engineer124 — COMMENTED** [review]
Great work! A few small comments

File: src/overlays/actors/ovl_En_Zot/z_en_zot.c (hunk 4 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Zot/z_en_zot.c:537]: This is the formatter's work.

**Review by engineer124 — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**